syne_tune/optimizer/schedulers/searchers/searcher.py

```diff
@@ -488,6 +501,11 @@ class GridSearcher(BaseSearcher):
         the full specification of the Hyperparameters, and the configurations
         generated is the combinations of these Hyperparameters.
         The specified hyperparameter must be Categorical for now.
+    num_samples: dict
```

Comment:
Can you add a comment, what will be used if you leave it to the default, i.e num_samples=None